### ISSUE TITLE
Fitting Performance of Sigmoid function

### ISSUE THREAD

**Issue by jjihoon** [itdxer/neupy#208]
Hi,

I created a new issue as I could not figure out the reason why the training performed so poorly in fitting to the output data. So, I trained the data based on the following condition:

x_train, x_test, y_train, y_test = train_test_split(
    data, target, test_size=0.20
)

cgnet = algorithms.Momentum(
connection=[
layers.Input(3),
layers.Sigmoid(2),
layers.Sigmoid(1),
],
step=0.6,
shuffle_data=True,
show_epoch=100,
train_end_signal=print_epoch_weight,
verbose=True,
)

cgnet.train(x_train, y_train, x_test, y_test, epochs=200)

As you can see in the code, I used Sigmoid function as an activation function for training. Then, I printed out predicted values derived from training in order to compare with output values.

![fitting performance](https://user-images.githubusercontent.com/24952529/36582139-9fc70ec0-18b3-11e8-9104-a8e5ec70f811.JPG)

Attached is the graph that shows output data values and three types of predicted values that are derived from training with 200, 400, and 600 epochs. As the graph shows, those predicted values do not fit to the shape of output data. 

I tried to figure out the cause of the bad performance but failed it. So, I wrote a post to ask for any advice about it. Maybe did I use an activation function that is not appropriate to apply to the output data? Or is there any restriction on the weight values which might kept the training from fitting to the output values?

Thanks

**Comment by itdxer** [comment]
It can be that your step size is too large. Try to run training with `0.1` or `0.05` instead of `0.6`

**Comment by jjihoon** [comment]
Thanks for the reply!

Like you said, I ran the algorithm with different step sizes, 0.001, 0.01, 0.05, and 0.1, and checked how its predicted values fit to the actual data. The result is shown below:

![fitting performance by difference step sizes](https://user-images.githubusercontent.com/24952529/36649348-c151a9d2-1adf-11e8-89d9-6f220b971ff7.jpg)

As the graph shows, it is difficult to say that applying smaller step sizes brought about meaningful improvement in fitting performance. Those predicted values still form somewhat linear shapes. Therefore, I wonder how come they are linearly shaped when the nonlinear Sigmoid function is applied to the algorithm as an activation function.

Thanks

**Comment by itdxer** [comment]
it's hard to tell without the data. can you share your data?

**Comment by jjihoon** [comment]
Sure, attached is the data that I used in the training.

There are four columns in the data, which are named Volatility, Strike Over Current Price, Drift, and Option Price. Those three columns of Volatility, Strike Over Current Price, Drift are used as input data while Option Price is used as an output data.

[EVanillaOption_Data_DriftFixed_701_Adjusted_OptionPrice_0.1_added.xlsx](https://github.com/itdxer/neupy/files/1757734/EVanillaOption_Data_DriftFixed_701_Adjusted_OptionPrice_0.1_added.xlsx)

Thanks.

**Comment by itdxer** [comment]
Looks like you haven't trained for long enough. That's what I get after 10,000 iterations

![](https://monosnap.com/file/DgkV0oCNEaqtVwuzduQ8aIFw5lotFC.png)

Also, it looks like `Volatility` and `Drift` variables are useless since they have exactly the same value, which means that you can remove them before the training and use single column.
